bahmnicore-api/src/test/java/org/bahmni/module/bahmnicore/forms2/mapper/FormDetailsMapperTest.java

```diff
@@ -0,0 +1,238 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+
+@PrepareForTest({FormUtil.class, FormType.class})
+@RunWith(PowerMockRunner.class)
+public class FormDetailsMapperTest {
+
+    private String formFieldPath = "FormName.2/1-0";
+    private String encounterUuid = "encounter-Uuid";
+    private String visitUuid = "visitUuid";
+    private String providerName = "Super Man";
+    private String providerUuid = "provider-uuid";
+    private String formName = "formName";
+    private int formVersion = 2;
+    private Date encounterDateTime = new Date();
+    private Date visitStartDateTime = new Date();
+
+    private Obs obs = mock(Obs.class);
+    private Encounter encounter = mock(Encounter.class);
+    private Visit visit = mock(Visit.class);
+    private User anotherCreator = mock(User.class);
+    private PersonName personName = mock(PersonName.class);
+
+    @Before
+    public void setUp() {
+
+        mockStatic(FormUtil.class);
+
+        when(obs.getEncounter()).thenReturn(encounter);
+        when(obs.getCreator()).thenReturn(anotherCreator);
+        when(obs.getFormFieldPath()).thenReturn(formFieldPath);
+        when(encounter.getVisit()).thenReturn(visit);
+        when(FormUtil.getFormNameFromFieldPath(formFieldPath)).thenReturn(formName);
+        when(FormUtil.getFormVersionFromFieldPath(formFieldPath)).thenReturn(formVersion);
+
+        when(encounter.getUuid()).thenReturn(encounterUuid);
+        when(encounter.getEncounterDatetime()).thenReturn(encounterDateTime);
+
+        when(visit.getUuid()).thenReturn(visitUuid);
+        when(visit.getStartDatetime()).thenReturn(visitStartDateTime);
+
+        when(anotherCreator.getPersonName()).thenReturn(personName);
+        when(personName.getFullName()).thenReturn(providerName);
+        when(anotherCreator.getUuid()).thenReturn(providerUuid);
+    }
+
+    @Test
+    public void shouldReturnFormDetailsFromGivenObsAndFormTypeOfFormBuilder() {
+
+        FormType formType = mock(FormType.class);
+        Whitebox.setInternalState(FormType.class, "FORM_BUILDER_FORMS", formType);
+        when(formType.get()).thenReturn("v2");
+
+        Collection<FormDetails> formDetailsCollection = FormDetailsMapper
+                .createFormDetails(singletonList(obs), formType);
+
+        assertEquals(1, formDetailsCollection.size());
+
+        FormDetails formDetails = formDetailsCollection.iterator().next();
+        assertEquals("v2", formDetails.getFormType());
+        assertEquals(formName, formDetails.getFormName());
+        assertEquals(formVersion, formDetails.getFormVersion());
+
+        verifyCommonData(formDetails);
+
+        verify(obs, times(2)).getFormFieldPath();
+        verifyStatic(VerificationModeFactory.times(1));
```

Comment:
What are we verifying?

Reply:
It verifies the next static method called once. Find the [doc](https://static.javadoc.io/org.powermock/powermock-api-mockito/1.6.2/org/powermock/api/mockito/PowerMockito.html#verifyStatic-org.mockito.verification.VerificationMode-)